search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalApiResponseBuilder.kt

```diff
@@ -0,0 +1,224 @@
+    @SuppressWarnings("LongParameterList")
+    fun buildEntitiesTemporalResponse(
+        entities: List<CompactedEntity>,
+        total: Int,
+        resourceUrl: String,
+        query: TemporalEntitiesQuery,
+        requestParams: MultiValueMap<String, String>,
+        mediaType: MediaType,
+        contexts: List<String>,
+        lang: String? = null
+    ): ResponseEntity<String> {
+        val ngsiLdDataRepresentation = parseRepresentations(requestParams, mediaType)
+        lang?.let { ngsiLdDataRepresentation.copy(languageFilter = it) }
+        val successResponse = buildQueryResponse(
+            entities.toFinalRepresentation(ngsiLdDataRepresentation),
+            total,
+            resourceUrl,
+            query.entitiesQuery.paginationQuery,
+            requestParams,
+            mediaType,
+            contexts
+        )
+
+        if (query.temporalQuery.hasLastN) { // if lastN > limit it throw an error earlier
+            return successResponse
+        }
+
+        val attributesWhoReachedLimit = getAttributesWhoReachedLimit(entities, query)
+
+        if (attributesWhoReachedLimit.isEmpty()) {
+            return successResponse
+        }
+
+        val range = getTemporalPaginationRange(attributesWhoReachedLimit, query)
+
+        val filteredEntities = entities.map { filterEntityInRange(it, range, query) }
+        return ResponseEntity.status(HttpStatus.PARTIAL_CONTENT).apply {
+            this.headers(
+                successResponse.headers
+            )
+            this.header(
+                HttpHeaders.CONTENT_RANGE,
+                getHeaderRange(range)
+            )
+        }.body(serializeObject(filteredEntities.toFinalRepresentation(ngsiLdDataRepresentation)))
+    }
+
+    fun buildEntityTemporalResponse(
+        entity: CompactedEntity,
+        query: TemporalEntitiesQuery,
+        mediaType: MediaType,
+        requestParams: MultiValueMap<String, String>,
+        contexts: List<String>,
+    ): ResponseEntity<String> {
+        val ngsiLdDataRepresentation = parseRepresentations(requestParams, mediaType)
+        val successResponseHeader = prepareGetSuccessResponseHeaders(mediaType, contexts)
+
+        if (query.temporalQuery.hasLastN) { // if lastN > limit it throw an error earlier
+            return successResponseHeader.body(
+                serializeObject(
+                    entity.toFinalRepresentation(ngsiLdDataRepresentation)
+                )
+            )
+        }
+
+        val attributesWhoReachedLimit = getAttributesWhoReachedLimit(listOf(entity), query)
+
+        if (attributesWhoReachedLimit.isEmpty()) {
+            return successResponseHeader.body(
+                serializeObject(
+                    entity.toFinalRepresentation(ngsiLdDataRepresentation)
+                )
+            )
+        }
+
+        val range = getTemporalPaginationRange(attributesWhoReachedLimit, query)
+        val filteredEntity = filterEntityInRange(entity, range, query)
```

Comment:
> * since there is a limit set in the SQL request, no data should be removed from the entity (this is why I don't really get the `filterEntityInRange` function)

Since the temporal data we receive can have different start time and endTime we can't just take the global least and most recent timestamp.
So i must get the least recent and most recent timestamp from what i call the discriminating attribute. (The first one to reach the limit)
Then i filter all the data that is not in this range